src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -391,4 +395,26 @@ private Boolean checkForValidInstitution(Integer institutionId) {
         return Objects.nonNull(existingId) && existingId > 0;
     }
 
+    public User findOrCreateUser(AuthUser authUser) {
+        User user;
+        // Ensure that the user is a registered DUOS user
+        try {
+            user = userDAO.findUserByEmail(authUser.getEmail());
+        } catch (NotFoundException nfe) {
+            User newUser = new User();
+            newUser.setEmail(authUser.getEmail());
+            newUser.setDisplayName(authUser.getName());
+            user = createUser(newUser);
+        }
+        // Ensure that the user is a registered SAM user
+        try {
+            samDAO.postRegistrationInfo(authUser);
+        } catch (ConsentConflictException cce) {
+          // no-op in the case of conflicts.
+        } catch (Exception e) {
+          throw new NotFoundException();
```

Comment:
I don't think we should be swallowing the exception and re-throwing as a not-found here. There could be network errors, or server exceptions that are best left to the resource class to handle. This pattern was originally in a resource class to begin with (see `SamResource`) so that layer figure out what kind of error to return to the client based on the type of exception thrown.

Reply:
Updated this logic to catch and log the exception instead of throwing a NotFoundException.